xen/x86: Force each VCPU to appear from a seperate package

VCPU's have no fixed CPU topology so make each VCPU appear to be on a
seperate CPU package.
diff --git a/arch/x86/xen/smp.c b/arch/x86/xen/smp.c
index a1e58e1..319f9e5 100644
--- a/arch/x86/xen/smp.c
+++ b/arch/x86/xen/smp.c
@@ -75,6 +75,7 @@ static void __cpuinit cpu_bringup(void)
 	cpu = smp_processor_id();
 	smp_store_cpu_info(cpu);
 	cpu_data(cpu).x86_max_cores = 1;
+	smp_num_siblings = 1;
 	set_cpu_sibling_map(cpu);
 
 	xen_setup_cpu_clockevents();
@@ -273,6 +274,8 @@ static void __init xen_smp_prepare_cpus(unsigned int max_cpus)
 		zalloc_cpumask_var(&per_cpu(cpu_core_map, i), GFP_KERNEL);
 		zalloc_cpumask_var(&per_cpu(cpu_llc_shared_map, i), GFP_KERNEL);
 	}
+	smp_num_siblings = 1;
+	boot_cpu_data.x86_max_cores = 1;
 	set_cpu_sibling_map(0);
 
 	if (xen_smp_intr_init(0))
